### PR TITLE
Add region-prompt modal dialog

### PR DESCRIPTION
**Pull request by WitLi07**
If no region is pre-configured, before redirecting to the login page, the region-select modal will show up for users to select their region.

### CODE REVIEW

File: app/index.html [hunk 1]
```diff
@@ -74,5 +74,29 @@ <h4 class="modal-title">Javascript Code Editor</h4>
...
+    <div>    
+        <div class="modal" tabindex="-1" id="regionModal" role="dialog" data-backdrop="false" aria-labelledby="regionModal" aria-hidden="true">
+            <div class="modal-dialog modal-dialog-flat">
+                <div class="modal-content">
+                    <div class="modal-header">
+                        <strong>Select A Region</strong>
+                        <button type="button" class="close" data-dismiss="modal" aria-label="Close"><span aria-hidden="true">&times;</span></button>
+                    </div>
+                    <div class="modal-body" id="regionModalBody">
+                            <button type="button" class="regionButton" value="mypurecloud.com">US Eest</button>
```
Comment (timsmithgenesys) [inline app/index.html:87]: Typo in "Eest"

Also, include the domain in each label like `US East (mypurecloud.com)` and `EU West (euw2.pure.cloud)`. Users may not know the proper designation of their region, but should be familiar with the URL they use to log in to the Genesys Cloud app.

**Comment by timsmithgenesys** [comment]
image

This close button on the modal doesn't do anything, nor should it. Selecting a region isn't optional and can't be canceled. The X button should be removed.

**Comment by timsmithgenesys** [comment]
image

Carry this title header through to the region selection screen. It doesn't have to be that actual UI element that's showing, but recreate it for the modal so it looks like the modal is appearing below it. Currently, it's unclear what you're being asked to choose a reason for.

image

Add some text below the heading and above the buttons that says something like "You must authenticate with Genesys Cloud to use this application. Please select your region to proceed." (using authenticate vs. authorize is intentional here as it's referring to the user authenticating with the login service; authorizing the app is implied.)

File: app/initializers/authenticators.js [hunk 2]
```diff
@@ -15,66 +16,112 @@ export default {
...
-		client.setPersistSettings(true, 'purecloud-dev-tools-auth');
-		client.setEnvironment(env);
-
 		// Authenticate
 		client
-			.loginImplicitGrant(oauthConfig.clientId, oauthConfig.redirect, { state: state })
-			.then((data) => {
-				// Store returned state
-				returnedState = data.state;
+		.loginImplicitGrant(oauthConfig.clientId, oauthConfig.redirect, { state: state })
```
Comment (timsmithgenesys) [inline app/initializers/authenticators.js:32]: Fix formatting in this block; the indentation is incorrect.

File: app/utils/purecloud-environment.js [hunk 3]
```diff
@@ -1,12 +1,9 @@
-const ENV_REG_EXP = /mypurecloud\.[a-z0-9.]{2,6}$|[a-z0-9]{4,5}.pure.cloud|inin[a-z0-9]{3}\.com|localhost/i;
 import config from '../config/environment';
 
 function purecloudEnvironmentTld() {
-	let env = ENV_REG_EXP.exec(window.location.hostname)[0];
 
-	if (env === 'localhost') {
-		env = 'mypurecloud.com';
-	}
+	let storage = window.localStorage;
```
Comment (timsmithgenesys) [inline app/utils/purecloud-environment.js:5]: This should use `app/services/storage-service.js`

File: app/services/purecloud.js [hunk 4]
```diff
@@ -92,6 +92,7 @@ export default Ember.Service.extend(Ember.Evented, {
 			.then(() => {
 				console.log('token destroyed');
 				window.localStorage.removeItem('purecloud_dev_tools_auth_auth_data');
+				window.localStorage.removeItem("environment");
```
Comment (timsmithgenesys) [inline app/services/purecloud.js:95]: This should use `app/services/storage-service.js`.

I believe the line above it is forcibly scrubbing the remembered auth token. Verify that the item key it's using is correct (it doesn't look right off the top of my head). That one needs to use window.localStorage directly as it's reaching outside the scope of the app itself trying to clear something set by the SDK.

File: app/index.html (hunk 1 above)
Comment (WitLi07) [reply app/index.html]: Fixed

File: app/utils/purecloud-environment.js (hunk 3 above)
Comment (WitLi07) [reply app/utils/purecloud-environment.js:5]: Talked about this. Decide to just use window.localStorage instead of storage-service.

**Review by timsmithgenesys — APPROVED** [review]

**Pull request merged**